src/cuda_packet.cpp

```diff
@@ -121,13 +125,67 @@ void jitc_cuda_render_gather_packet(const Variable *v, const Variable *ptr,
     }
 }
 
+/**
+ * Render the code required to scatter reduce a packet of variables.
+ */
+void jitc_cuda_render_scatter_reduce_packet(const Variable *v,
+                                            const Variable *ptr,
+                                            const Variable *index,
+                                            const Variable *mask) {
+    bool is_masked = !mask->is_literal() || mask->literal != 1;
+    PacketScatterData *psd = (PacketScatterData *) v->data;
+    const std::vector<uint32_t> &values = psd->values;
+    const Variable *v0 = jitc_var(values[0]);
+
+    uint32_t count = (uint32_t) values.size(),
+             tsize = type_size[v0->type];
+
+    put("    {\n");
+
+    fmt("        .reg.f16x2 %tmp;\n");
+
+    fmt("        mad.wide.$t %rd3, $v, $u, $v;\n",
+        index, index, tsize, ptr);
+
+    if (v0->type != (uint32_t)VarType::Float16)
+        jitc_fail("Packeted scatter reductions are only supported with f16 "
+                  "variables.");
+
+    const char *op = reduce_op_name[(uint32_t) psd->op];
+
+    if (count % 2 == 0) {
```

Comment:
I think it would be great to expand this to use the `.v2, .v4, .v8` variants of `red`, like @wjakob suggested.